app/src/androidTest/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapterTest.kt

```diff
@@ -0,0 +1,69 @@
+package net.aiscope.gdd_app.ui.metadata
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.io.InputStream
+
+@RunWith(AndroidJUnit4::class)
+class SampleImagesAdapterTest {
+    private val testContext = InstrumentationRegistry.getInstrumentation().context
+    private val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
+    private val cacheDir = targetContext.cacheDir
+    private lateinit var targetFile: File
+
+    @Before
+    fun setup() {
+        targetFile = File(targetContext.filesDir, "test.bmp");
+        targetFile.createNewFile()
+
+        val stream = testContext.assets.open("w3c_home.bmp");
+        stream.toFile(targetFile)
+    }
+
+
+    @Test
+    fun shouldWriteToCache() {
+        val b = runBlocking {
+            decodeSampledBitmapAndCache(targetFile, 20, 20, cacheDir)
+        }
+
+        assertTrue(cacheDir.isDirectory)
+        val cachedFiles = cacheDir.listFiles()
+
+        assertTrue(cachedFiles.any { file -> file.name.equals("test_20x20.bmp", true) })
+        assertTrue(b.height < 50)
+        assertTrue(b.width < 50)
+        assertTrue(b.height > 20)

```

Comment:
Should we assert equality instead? Or do you think that this would couple the test too much to the implementation details? Either way, 20 would be a valid dimension as well. so this should be `>=`, no?

Reply:
Well I don't want to pin myself down to a certain number as a result since that's not explicitly expected. You're right on the equality though, adjusted (even though it does not occur with this specific implementation ;))

Reply:
Fair enough. Do you think that it would make sense to have a parametrised test here though? To catch more edge cases.

Reply:
Well these just test basic functionality. I am sure we _could_ add more tests for edge cases, but that sort of goes for the entire code base ;)

I don't think it's worth the effort right now